WordPressApi/WPComOAuthController.m

```diff
@@ -129,18 +125,17 @@ - (void)viewDidAppear:(BOOL)animated {
     if (userAgent) {
         [request setValue:userAgent forHTTPHeaderField:@"User-Agent"];
     }
-    if (_username && _password) {
-        NSString *request_body = [NSString stringWithFormat:@"log=%@&pwd=%@&redirect_to=%@",
-                                  [self stringByUrlEncodingString:_username],
-                                  [self stringByUrlEncodingString:_password],
-                                  [self stringByUrlEncodingString:queryUrl]];
+    if (_username && _authToken) {
+        NSString *requestBody = [NSString stringWithFormat:@"%@=%@&%@=%@&%@=%@",
+                                 @"log", [self stringByUrlEncodingString:_username],
+                                 @"pwd", [NSString string],
```

Comment:
I'm sorry to say that yes, without an empty password field the request is expected to fail, even if you send the Bearer token in the header.

Thanks Diego!